pkg/certificate/controller_test.go

```diff
@@ -126,15 +127,21 @@ var _ = Describe("Certificates controller", func() {
 		var (
 			previousResult, currentResult reconcile.Result
 			previousTLS, currentTLS       TLS
+			backToTheFuture               = func(step string, future time.Duration) {
```

Comment:
iiuc, this function has an isolation problem, because if you run it twice the "now" changes, this can be confusing since the "now" is not fixed...
can you set the "now" relative to a fixed "now" that is declared outside of this function?

Reply:
This is the whole purpose, calling it multiple time throw you to the future.

Reply:
let me explain with example: 
- first call `backToTheFuture("Reconcile for the first time", 0)`: mgr.now() and my watch have 0 difference
- second call `backToTheFuture("Reconcile in the middle of service cert deadline", serviceCertDuration/2)`: mgr.now() adds `duration/2` so it is after my watch, so far so good.
- third call `backToTheFuture("Reconcile at service cert rotation deadline", currentResult.RequeueAfter)`: mgr.now() adds `duration/2+currentResult.RequeueAfter`, and not `currentResult.RequeueAfter`...

in other words,
```
now = mgr.now().Add(future)
mgr.now = func() time.Time { return now }
```
so now always adds to the last mgr.now that was set earlier... this is what I mean by isolation issue.

Reply:
I still don't get it, this is a time machine and the offset is relative to the time you have moved to.